WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -468,6 +537,27 @@ class WeeklyRoundupBackgroundTask: BackgroundTask {
         operationQueue.addOperation(completionOperation)
     }
 
+    // MARK: - Helpers
+
+    private func scheduleDynamicNotification(site: Blog, stats: StatsSummaryData, completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
What do you think we pass the blog instance's `NSManagedObjectID` as argument and let the function query the blog instance itself? That'll prevent this function from being called from a queue that's not bound to the Blog argument.

Reply:
Resolved in 89e99ffc8e986db1afebd314e30007ebfed8da8d